### PR TITLE
solution

### PR DESCRIPTION
**Pull request by IhorRome**

### CODE REVIEW

**Review by andriiromaniuk — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -31,17 +31,15 @@
...
 
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-war-plugin</artifactId>
-                <configuration>
-                    <webXml>src/main/webapp/web.xml</webXml>
-                </configuration>
-            </plugin>
```
Comment (MaliukDaria) [inline pom.xml:44]: Why did you remove this plugin?

File: src/main/java/mate/controller/AddDriverToCarController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.service.CarService;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/car/add-driver-to-car")
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/AddDriverToCarController.java:15]:
```suggestion
@WebServlet(urlPatterns = "/cars/drivers/add")
```

Look at URLs naming rules in the checklist

File: src/main/java/mate/controller/AddDriverToCarController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        Long driverId = Long.valueOf(request.getParameter("driver_id"));
+        Long carId = Long.valueOf(request.getParameter("car_id"));
+        Car car = carService.get(carId);
+        Driver driver = driverService.get(driverId);
+        carService.addDriverToCar(driver, car);
+        request.getRequestDispatcher("/WEB-INF/views/car/add-driver.jsp")
+                .forward(request, response);
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/AddDriverToCarController.java:38]:
```suggestion
        resp.sendRedirect("/car/add-driver-to-car");
```

Its better to use sendRedirect here

File: src/main/java/mate/controller/CreateCarController.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        Long id = Long.valueOf(req.getParameter("manufacturer_id"));
+        Manufacturer manufacturer = manufacturerService.get(id);
+        Car car = new Car(model, manufacturer);
+        carService.create(car);
+        req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/CreateCarController.java:37]: same, its better to use sendRedirect here

File: src/main/java/mate/controller/CreateDriverController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String driverName = req.getParameter("name");
+        String driverLicenseNumber = req.getParameter("license_number");
+        Driver driver = new Driver(driverName, driverLicenseNumber);
+        driverService.create(driver);
+        req.getRequestDispatcher("/WEB-INF/views/drivers/create.jsp").forward(req, resp);
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/CreateDriverController.java:32]: same, its better to use sendRedirect here

File: src/main/java/mate/controller/CreateManufacturerController.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String manufacturerName = req.getParameter("name");
+        String manufacturerCountry = req.getParameter("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturerService.create(manufacturer);
+        req.getRequestDispatcher("/WEB-INF/views/manufacturer/create.jsp").forward(req, resp);
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/CreateManufacturerController.java:32]: same, its better to use sendRedirect here

File: src/main/java/mate/util/ConnectionUtil.java [hunk 7]
```diff
@@ -25,4 +26,8 @@ public static Connection getConnection() throws SQLException {
         dbProperties.setProperty("password", PASSWORD);
         return DriverManager.getConnection(URL, dbProperties);
     }
+
+    public static void main(String[] args) throws SQLException {
+        Connection connection = ConnectionUtil.getConnection();
+    }
```
Comment (MaliukDaria) [inline src/main/java/mate/util/ConnectionUtil.java:32]: do we need this here?

File: src/main/webapp/WEB-INF/web.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
+                      http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
+         version="4.0">
+</web-app>
```
Comment (MaliukDaria) [inline src/main/webapp/WEB-INF/web.xml:7]: no newline at the end of file

File: src/main/java/mate/controller/DisplayDriversController.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Driver;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/driver/all")
```
Comment (MaliukDaria) [inline src/main/java/mate/controller/DisplayDriversController.java:14]:
```suggestion
@WebServlet(urlPatterns = "/drivers")
```

**Review by MaliukDaria — CHANGES_REQUESTED** [review]
